hack/deploy_with_helm.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
+
+METHOD=install
+ONLY_DB=
+NO_AUTH=
+
+# Use external getopt for long options
+options=$(getopt -o adh --long only-db,no-auth,help -n "$0" -- "$@")
+eval set -- "$options"
+
+while true; do
+  case "$1" in
+    -a|--only-db) ONLY_DB="--set flightctl.server.enabled=false" ; shift ;;
+    -d|--no-auth) NO_AUTH="--set flightctl.keycloak.enabled=false"; shift ;;
+    -h|--help) echo "Usage: $0 [--only-db] [--no-auth]"; exit 0 ;;
+    --) shift; break ;;
+    *) echo "Invalid option: $1" >&2; exit 1 ;;
+  esac
+done
+
+# if we have an existing deployment, try to upgrade it instead
+if helm list -A | grep flightctl > /dev/null; then
+  METHOD=upgrade
+fi
+
+# helm expects the namespaces to exist, and creating namespaces
+# inside the helm charts is not recommended.
+kubectl create namespace flightctl-external 2>/dev/null || true
+kubectl create namespace flightctl-internal 2>/dev/null || true
+
+# if we are only deploying the database, we don't need inject the server container
+if [ -z "$ONLY_DB" ]; then
+  # In github CI, kind gets confused and tries to pull the image from docker instead
+  # of podman, so if regular docker-image fails we need to:
+  #   * save it to OCI image format
+  #   * then load it into kind
+  kind load docker-image localhost/flightctl-server:latest || \
+    podman save localhost/flightctl-server:latest -o flightctl-server.tar && \
+    kind load image-archive flightctl-server.tar
+fi
+
+# if we need another database image we can set it with PGSQL_IMAGE (i.e. arm64 images)
+if [ ! -z "$PGSQL_IMAGE" ]; then
+  DB_IMG="--set flightctl.db.image=${PGSQL_IMAGE}"
+fi
+
+helm ${METHOD} --values ./deploy/helm/flightctl/values.kind.yaml ${ONLY_DB} ${NO_AUTH} ${DB_IMG} flightctl ./deploy/helm/flightctl/
+
+"${SCRIPT_DIR}"/wait_for_postgres.sh
+
+# Make sure the database is usable from the unit tests
+DB_POD=$(kubectl get pod -n flightctl-internal -L flightctl.service=flightctl-db --no-headers -o custom-columns=":metadata.name")
+kubectl exec -n flightctl-internal "${DB_POD}" -- psql -c 'ALTER ROLE admin WITH SUPERUSER'
+kubectl exec -n flightctl-internal "${DB_POD}" -- createdb admin 2>/dev/null|| true
+
+mkdir -p  ~/.flightctl/certs
```

Comment:
evertything after this line could be moved to it's own script (useful if you need to re-extract credentials from a running server volume).

i.e.
`./hack/get_flightctl_server_creds.sh`